### PR TITLE
FUSETOOLS-1951 - First cut of templates

### PR DESCRIPTION
**Pull request by bfitzpat**
Adds new directories for templates to be used in the new project wizard

### CODE REVIEW

File: templates/medium/cxf-contract-first/medium-fuse-cxf-contract-first-blueprint/pom.xml [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+			<distribution>repo</distribution>
+		</license>
+	</licenses>
+	<properties>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+		<version.maven-bundle-plugin>2.3.7</version.maven-bundle-plugin>
+		<maven.compiler.target>1.7</maven.compiler.target>
+		<maven.compiler.source>1.7</maven.compiler.source>
+		<version.maven-surefire-plugin>2.15</version.maven-surefire-plugin>
+		<skipTests>true</skipTests>
```
Comment (apupier) [inline templates/medium/cxf-contract-first/medium-fuse-cxf-contract-first-blueprint/pom.xml]: Why skipping the tests by default?

File: templates/medium/cxf-contract-first/medium-fuse-cxf-contract-first-blueprint/pom.xml [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+					<instructions>
+						<Export-Package>ws.camel.mycompany.com</Export-Package>
+					</instructions>
+				</configuration>
+			</plugin>
+
+			<!-- this plugin will use the fabric.* properties to configure its behaviour 
+				see the Properties section here: http://fabric8.io/gitbook/mavenPlugin.html -->
+			<plugin>
+				<groupId>io.fabric8</groupId>
```
Comment (apupier) [inline templates/medium/cxf-contract-first/medium-fuse-cxf-contract-first-blueprint/pom.xml]: does it work for you?
Form my experience currently, the fabric8 are not well handled as m2e connectors and neither on Windows

File: templates/medium/cxf-contract-first/medium-fuse-cxf-contract-first-spring/pom.xml [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+			<url>http://www.apache.org/licenses/LICENSE-2.0.html</url>
+			<distribution>repo</distribution>
+		</license>
+	</licenses>
+	<properties>
+		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+		<version.maven-bundle-plugin>2.3.7</version.maven-bundle-plugin>
+		<maven.compiler.target>1.7</maven.compiler.target>
+		<maven.compiler.source>1.7</maven.compiler.source>
+		<version.maven-surefire-plugin>2.15</version.maven-surefire-plugin>
```
Comment (apupier) [inline templates/medium/cxf-contract-first/medium-fuse-cxf-contract-first-spring/pom.xml]: the current version is 2.19.1, might worth to upgrade it

File: templates/blank/blank-blueprint/pom.xml [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+  <groupId>com.mycompany</groupId>
+  <artifactId>camel-blueprint</artifactId>
+  <version>1.0.0-SNAPSHOT</version>
+  <packaging>bundle</packaging>
+
+  <name>Camel Blueprint Quickstart</name>
+  <description>Empty Camel Blueprint Example</description>
+
+  <licenses>
```
Comment (lhein) [inline templates/blank/blank-blueprint/pom.xml]: Should we really have ASL in our templates?

File: templates/blank/blank-blueprint/pom.xml [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+            <url>https://repo.fusesource.com/nexus/content/groups/public</url>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+            <releases>
+                <enabled>true</enabled>
+                <updatePolicy>never</updatePolicy>
+            </releases>
+        </repository>
+		<repository>
```
Comment (lhein) [inline templates/blank/blank-blueprint/pom.xml]: Indention wrong

File: templates/blank/blank-blueprint/pom.xml [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+            <url>https://repo.fusesource.com/nexus/content/groups/public</url>
+            <snapshots>
+                <enabled>false</enabled>
+            </snapshots>
+            <releases>
+                <enabled>true</enabled>
+                <updatePolicy>never</updatePolicy>
+            </releases>
+        </pluginRepository>
+		<pluginRepository>
```
Comment (lhein) [inline templates/blank/blank-blueprint/pom.xml]: indention

File: templates/medium/cxf-contract-first/medium-fuse-cxf-contract-first-spring/pom.xml [hunk 7]
```diff
@@ -0,0 +1,276 @@
...
+			</plugin>
+
+			<plugin>
+				<groupId>org.apache.felix</groupId>
+				<artifactId>maven-bundle-plugin</artifactId>
+				<version>${version.maven-bundle-plugin}</version>
+				<extensions>true</extensions>
+				<configuration>
+					<instructions>
+						<Export-Package>ws.camel.mycompany.com</Export-Package>
```
Comment (apupier) [inline templates/medium/cxf-contract-first/medium-fuse-cxf-contract-first-spring/pom.xml]: we package as bundle for spring based project too?

File: templates/blank/blank-blueprint/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 8]
```diff
@@ -0,0 +1,38 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    JBoss, Home of Professional Open Source
+    Copyright 2014, Red Hat, Inc. and/or its affiliates, and individual
+    contributors by the @authors tag. See the copyright.txt in the
+    distribution for a full listing of individual contributors.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
```
Comment (lhein) [inline templates/blank/blank-blueprint/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: Do we really need the ASL here?

**Comment by lhein** [comment]
Please remove all src/META-INF/MANIFEST.MF from all the templates.

**Comment by lhein** [comment]
also all src/main/java/META-INF

**Comment by bfitzpat** [comment]
Some of these work and some do not - it's a work in progress. Making them ALL work is the goal of course. But I ran into issues with getting some of the tests to run - thus what you see here. 

But I figured it was easier for other folks (such as yourselves) to be able to hack on them if we got them into github.

**Pull request merged**